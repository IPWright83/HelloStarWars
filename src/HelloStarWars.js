import React, { Component } from "react";

class HelloStarWars extends React.PureComponent {

	constructor(props) {
		super(props);
		this.state = {};
	}

	async componentDidMount() {
		const id = this.props.id;

		try {
			const response = await fetch(`https://swapi.co/api/people/${id}`);
			const { name } = await response.json();
			this.setState({ name });
		} catch (e) {
			this.setState({ name: "!dlroW... Oh dear, something went wrong" });
		}
	}

	render() {
		const name = this.state.name || "...";
		return(
			<h1>Hello {name}!</h1>
		);
	}
}

export default HelloStarWars;