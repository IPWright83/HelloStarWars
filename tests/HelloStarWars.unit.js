import React from "react";
import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
import enzyme from "enzyme";
import { mount } from "enzyme";

import HelloStarWars from "../src/HelloStarWars";

// Uses jest-fetch-mock setup within the setupJest.js file
enzyme.configure({ adapter: new Adapter() });

xdescribe("HelloStarWars", () => {

	it("Renders with a place holder", () => {
		const result = renderer
			.create(<HelloStarWars id="1" />)
			.toJSON();

		expect(result).toMatchSnapshot();
	});

	describe("ComponentDidMount", () => {

		beforeEach(() => {
			fetch.resetMocks();
		});

		it("Obtains result and sets data correctly", async () => {
			fetch.mockResponseOnce(JSON.stringify({ name: 'Jedi Master' }));

			const wrapper = await mount(<HelloStarWars id="1" />);
			const instance = wrapper.instance();
			
			await instance.componentDidMount();

			wrapper.update();
			await Promise.resolve();

			expect(wrapper).toMatchSnapshot();
		});

		it("Handles an error in the web request correctly", async () => {
			const wrapper = await mount(<HelloStarWars id="foobar" />);
			const instance = wrapper.instance();

			fetch.mockReject(new Error("Invalid ID"));
			await instance.componentDidMount();

			wrapper.update();
			expect(wrapper).toMatchSnapshot();
		});
	});
});