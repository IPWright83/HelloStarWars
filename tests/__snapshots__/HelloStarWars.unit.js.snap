// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloStarWars ComponentDidMount Handles an error in the web request correctly 1`] = `
<HelloStarWars
  id="foobar"
>
  <h1>
    Hello 
    !dlroW... Oh dear, something went wrong
    !
  </h1>
</HelloStarWars>
`;

exports[`HelloStarWars ComponentDidMount Obtains result and sets data correctly 1`] = `
<HelloStarWars
  id="1"
>
  <h1>
    Hello 
    Jedi Master
    !
  </h1>
</HelloStarWars>
`;

exports[`HelloStarWars Renders with a place holder 1`] = `
<h1>
  Hello 
  ...
  !
</h1>
`;
